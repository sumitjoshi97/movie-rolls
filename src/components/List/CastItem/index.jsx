import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class CastItem extends PureComponent {
  static propTypes = {
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    role: PropTypes.string
  }

  render() {
    const { poster, name, role } = this.props

    return (
      <div className='list-item'>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt={name}
          className='list-item__img'
        />
        <div className='list__item__name'>{name}</div>
        <div className='list__item__name__sub'>{role}</div>
      </div>
    )
  }
}
