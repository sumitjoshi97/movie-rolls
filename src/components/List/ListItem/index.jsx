import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import * as actions from '../../../actions'

class ListItem extends PureComponent {
  static propTypes = {
    pageId: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    setType: PropTypes.func.isRequired
  }

  render() {
    const { pageId, type, poster, name, setType } = this.props

    return (
      <div className='list-item'>
        <Link to={`/details/${pageId}`} onClick={() => setType(type)}>
          <img
            src={`https://image.tmdb.org/t/p/w185/${poster}`}
            alt={name}
            className='list-item__img'
          />
          <div className='list__item__name'>{name}</div>
        </Link>
        )
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setType: type => dispatch(actions.setItemType(type))
})

export default connect(
  null,
  mapDispatchToProps
)(ListItem)
