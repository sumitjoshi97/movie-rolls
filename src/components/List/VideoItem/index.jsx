import React from 'react'
import PropTypes from 'prop-types'

const VideoItem = ({ video, name, category }) => (
  <div className='list-item'>
    <iframe
      className='list-item__video'
      title='1'
      width='280'
      src={`https://www.youtube.com/embed/${video}`}
    />
    <div className='list-item__name'>{name}</div>
    <div className='list-item__name__sub'>{category}</div>
  </div>
)

VideoItem.propTypes = {
  video: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string
}

export default VideoItem
