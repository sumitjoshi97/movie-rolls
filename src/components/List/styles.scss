.list {
  padding: 6rem 6rem 0 6rem;
  background-color: #1a1a1a;
  &__header {
    text-transform: capitalize;
    padding-bottom: 1rem;
  }
}

.list-item {
  display: inline-block;
  padding: 2rem;
  &__img {
    height: 18rem;
    background: #000;
    width: 14rem;
  }

  &__video {
    display: inline-block;
    height: 18rem;
    outline: none;
    border: none;
  }

  &__name {
  }
}
