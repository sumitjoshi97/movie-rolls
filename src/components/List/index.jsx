import React, { PureComponent } from 'react'
import Slider from 'react-slick'

import ListItem from './ListItem'
import CastItem from './CastItem'
import VideoItem from './VideoItem'
import NextArrow from '../Arrows/Next'
import PrevArrow from '../Arrows/Prev'

import './styles.scss'

export default class List extends PureComponent {
  renderList = () => {
    const { type, items } = this.props

    switch (type) {
      case 'cast':
        return items.map(item => (
          <CastItem
            key={item.id}
            name={item.name}
            poster={item.profile_path}
            role={item.character}
          />
        ))

      case 'video':
        return items.map(item => (
          <VideoItem
            key={item.id}
            name={item.name}
            video={item.key}
            category={item.type}
          />
        ))

      //for movies / tv shows
      default:
        return items.map(item => (
          <ListItem
            key={item.id}
            type={type}
            pageId={item.id}
            name={item.title}
            rating={item.vote_average}
            poster={item.poster_path}
          />
        ))
    }
  }

  render() {
    const sliderOptions = {
      arrows: true,
      dots: false,
      speed: 400,
      slidesToShow: this.props.type === 'video' ? 4 : 7,
      swipeToSlide: true,
      infinite: false,
      nextArrow: <NextArrow />,
      prevArrow: <PrevArrow />
    }
    const { name } = this.props
    return (
      <div className='list'>
        <h2 className='list__header'>{name}</h2>
        <Slider {...sliderOptions}>{this.renderList()}</Slider>
      </div>
    )
  }
}
