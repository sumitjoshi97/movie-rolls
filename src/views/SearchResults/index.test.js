import React from 'react'
import { shallow } from 'enzyme'

import { SearchResults } from './index'
import List from '../../components/List'

describe('SearchResults component unit tests', () => {
  const getSearchResults = jest.fn()
  const match = { params: { searchQuery: 'test' } }

  describe('when search results are not null', () => {
    const searchResults = {
      results: [
        {
          id: '0',
          media_type: 'movie'
        },
        {
          id: '0',
          media_type: 'tv'
        }
      ]
    }
    const props = {
      getSearchResults,
      searchResults,
      match
    }

    const searchWrapper = shallow(<SearchResults {...props} />)

    test('should render properly', () => {
      expect(searchWrapper).toMatchSnapshot()
    })

    test('should call getSearchResults fn. with `searchQuery`', () => {
      expect(getSearchResults).toHaveBeenCalledWith(match.params.searchQuery)
    })

    test('should render 2 lists', () => {
      expect(searchWrapper.find(List)).toHaveLength(2)
    })
  })

  describe('when searchResults are null', () => {
    const props = {
      getSearchResults,
      searchResults: null,
      match
    }

    const searchWrapper = shallow(<SearchResults {...props} />)

    test('should render `error/sorry-text`', () => {
      expect(searchWrapper.find('h2.sorry-text'))
    })
  })
})
