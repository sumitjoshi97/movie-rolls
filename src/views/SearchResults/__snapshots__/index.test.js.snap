// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component unit tests when search results are not null should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    getSearchResults={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
          Array [
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    match={
      Object {
        "params": Object {
          "searchQuery": "test",
        },
      }
    }
    searchResults={
      Object {
        "results": Array [
          Object {
            "id": "0",
            "media_type": "movie",
          },
          Object {
            "id": "0",
            "media_type": "tv",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-results__header"
        >
          <h2
            className="search-results__header__primary"
          >
            Search Results
          </h2>
          <h4
            className="search-results__header__secondary"
          >
            test
          </h4>
        </div>,
        <div
          className="search-results__list"
        >
          <React.Fragment>
            <List
              items={
                Array [
                  Object {
                    "id": "0",
                    "media_type": "movie",
                  },
                ]
              }
              name="Movies"
              type="movie"
            />
            <List
              items={
                Array [
                  Object {
                    "id": "0",
                    "media_type": "tv",
                  },
                ]
              }
              name="TV Shows"
              type="tv"
            />
          </React.Fragment>
        </div>,
      ],
      "className": "search-results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="search-results__header__primary"
            >
              Search Results
            </h2>,
            <h4
              className="search-results__header__secondary"
            >
              test
            </h4>,
          ],
          "className": "search-results__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search Results",
              "className": "search-results__header__primary",
            },
            "ref": null,
            "rendered": "Search Results",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
              "className": "search-results__header__secondary",
            },
            "ref": null,
            "rendered": "test",
            "type": "h4",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <React.Fragment>
            <List
              items={
                Array [
                  Object {
                    "id": "0",
                    "media_type": "movie",
                  },
                ]
              }
              name="Movies"
              type="movie"
            />
            <List
              items={
                Array [
                  Object {
                    "id": "0",
                    "media_type": "tv",
                  },
                ]
              }
              name="TV Shows"
              type="tv"
            />
          </React.Fragment>,
          "className": "search-results__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "movie",
                    },
                  ]
                }
                name="Movies"
                type="movie"
              />,
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "tv",
                    },
                  ]
                }
                name="TV Shows"
                type="tv"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "items": Array [
                  Object {
                    "id": "0",
                    "media_type": "movie",
                  },
                ],
                "name": "Movies",
                "type": "movie",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "items": Array [
                  Object {
                    "id": "0",
                    "media_type": "tv",
                  },
                ],
                "name": "TV Shows",
                "type": "tv",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-results__header"
          >
            <h2
              className="search-results__header__primary"
            >
              Search Results
            </h2>
            <h4
              className="search-results__header__secondary"
            >
              test
            </h4>
          </div>,
          <div
            className="search-results__list"
          >
            <React.Fragment>
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "movie",
                    },
                  ]
                }
                name="Movies"
                type="movie"
              />
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "tv",
                    },
                  ]
                }
                name="TV Shows"
                type="tv"
              />
            </React.Fragment>
          </div>,
        ],
        "className": "search-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="search-results__header__primary"
              >
                Search Results
              </h2>,
              <h4
                className="search-results__header__secondary"
              >
                test
              </h4>,
            ],
            "className": "search-results__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search Results",
                "className": "search-results__header__primary",
              },
              "ref": null,
              "rendered": "Search Results",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "search-results__header__secondary",
              },
              "ref": null,
              "rendered": "test",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <React.Fragment>
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "movie",
                    },
                  ]
                }
                name="Movies"
                type="movie"
              />
              <List
                items={
                  Array [
                    Object {
                      "id": "0",
                      "media_type": "tv",
                    },
                  ]
                }
                name="TV Shows"
                type="tv"
              />
            </React.Fragment>,
            "className": "search-results__list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <List
                  items={
                    Array [
                      Object {
                        "id": "0",
                        "media_type": "movie",
                      },
                    ]
                  }
                  name="Movies"
                  type="movie"
                />,
                <List
                  items={
                    Array [
                      Object {
                        "id": "0",
                        "media_type": "tv",
                      },
                    ]
                  }
                  name="TV Shows"
                  type="tv"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "items": Array [
                    Object {
                      "id": "0",
                      "media_type": "movie",
                    },
                  ],
                  "name": "Movies",
                  "type": "movie",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "items": Array [
                    Object {
                      "id": "0",
                      "media_type": "tv",
                    },
                  ],
                  "name": "TV Shows",
                  "type": "tv",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
